### PR TITLE
Metadata retained when copying from Word in Windows

### LINKED ISSUE #9002: CKEditor 5 copy “Font Definitions” from word document
When I copy some text from my word document (.docx) some "Font Definitions" are copied. For example:

Word document:
image

### ✔️ Expected result

image

### ❌ Actual result

_What is the actual result of the above steps?_
image

## 📃 Other details

* Browser: Chrome
* OS: Windows 10
We have the same issue in the demo documentation (https://ckeditor.com/docs/ckeditor5/latest/features/pasting/paste-from-word.html). 
With CKEditor 4, works fine.

---

If you'd like to see this fixed sooner, add a 👍 reaction to this post.

### LINKED ISSUE #15333: Some metadata is retained alongside the content when copying from a Word file on Windows
## 📝 Provide detailed reproduction steps (if any)

1. On a Widnows machine copy all of the contents of the attached Word file:

    [CK EDITOR TEST.docx](https://github.com/ckeditor/ckeditor5/files/13308273/CK.EDITOR.TEST.docx)

2. Paste it into the [Paste from Office demo](https://ckeditor.com/docs/ckeditor5/latest/features/pasting/paste-from-office.html#demo) and the [Paste from Office enhanced demo](https://ckeditor.com/docs/ckeditor5/latest/features/pasting/paste-from-office-enhanced.html#demo)

### ✔️ Expected result

Only the contents of the file should be copied, as is the case on macOS: 

https://github.com/ckeditor/ckeditor5/assets/72079603/43002dd5-be06-4da6-97c8-5a895838bca7

### ❌ Actual result

On Windows, in the regular PfO demo, some metadata is retained when copying the file:

https://github.com/ckeditor/ckeditor5/assets/72079603/3ed3490c-d66f-414e-b9e3-ac942adf4e86

Whereas in the PfOE demo, the contents of the file do not get copied at all and you can see a `<span>` with the attribute `data-ck-unsafe-element` when you inspect the editable:

https://github.com/ckeditor/ckeditor5/assets/72079603/a0c25e4a-5a7a-42ca-845a-cadb86a7c3ed

### ❓ Possible solution

I'm also including the resulting `text/html` returned by [Clipboard Inpector](https://evercoder.github.io/clipboard-inspector/) when copying and pasting the file on Windows:

[paste_text_html.txt](https://github.com/ckeditor/ckeditor5/files/13308286/paste_text_html.txt)

It's possible that the metadata that is being copied is treated as an HTML comment (it starts with `<!--` and ends with `-->`, so perhaps that's why it's being retained.

## 📃 Other details

* Browser: cross-browser
* OS: Windows 11

---

If you'd like to see this fixed sooner, add a 👍 reaction to this post.

### PR DESCRIPTION
**Pull request by mmotyczynska**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/contributing/git-commit-message-convention.html))

Fix (paste-from-office): Content from Word document should be pasted correctly on Windows systems. Closes #15333. Closes #9002.

---

### Additional information

_For example – encountered issues, assumptions you had to make, other affected tickets, etc._

### CODE REVIEW

**Review by arkflpc — APPROVED** [review]

File: packages/ckeditor5-paste-from-office/src/filters/parse.ts [hunk 1]
```diff
@@ -29,6 +29,11 @@ export function parseHtml( htmlString: string, stylesProcessor: StylesProcessor
 	// Remove Word specific "if comments" so content inside is not omitted by the parser.
 	htmlString = htmlString.replace( /<!--\[if gte vml 1]>/g, '' );
 
+	// Clean the <head> section of MS Windows specific tags. See #15333.
```
Comment (niegowski) [inline packages/ckeditor5-paste-from-office/src/filters/parse.ts:32]: Please use the full issue link.

File: packages/ckeditor5-paste-from-office/src/filters/parse.ts [hunk 2]
```diff
@@ -29,6 +29,11 @@ export function parseHtml( htmlString: string, stylesProcessor: StylesProcessor
...
+	// The regular expression matches tag <o:SmartTagType> with optional attributes (with or without values),
+	// optional white spaces after attributes and optional `/` before closing.
```
Comment (niegowski) [inline packages/ckeditor5-paste-from-office/src/filters/parse.ts:34]:
```suggestion
	// The regular expression matches the <o:SmartTagType> tag with optional attributes (with or without values).
```

File: packages/ckeditor5-paste-from-office/src/filters/parse.ts [hunk 3]
```diff
@@ -29,6 +29,11 @@ export function parseHtml( htmlString: string, stylesProcessor: StylesProcessor
...
+	htmlString = htmlString.replace( /<o:SmartTagType(?:\s+[^\s>]+(?:="[^"]*")?)*\s*\/?>/gi, '' );
```
Comment (niegowski) [inline packages/ckeditor5-paste-from-office/src/filters/parse.ts:35]:
```suggestion
	htmlString = htmlString.replace( /<o:SmartTagType(?:\s+[^\s>=]+(?:="[^"]*")?)*\s*\/?>/gi, '' );
```

File: packages/ckeditor5-paste-from-office/tests/filters/parse.js [hunk 4]
```diff
@@ -172,6 +172,68 @@ describe( 'PasteFromOffice - filters', () => {
...
+							'<p>foo</p>' +
+						'</body>' +
+					'</html>';
+					const { body, bodyString } = parseHtml( html );
+
+					expect( body ).to.instanceof( DocumentFragment );
+
+					expect( body.childCount ).to.equal( 1 );
+
+					expect( bodyString ).to.equal( '<p>foo</p>' );
```
Comment (niegowski) [inline packages/ckeditor5-paste-from-office/tests/filters/parse.js:215]:
```suggestion
					expect( body ).to.instanceof( DocumentFragment );
					expect( body.childCount ).to.equal( 1 );
					expect( bodyString ).to.equal( '<p>foo</p>' );
```

File: packages/ckeditor5-paste-from-office/tests/filters/parse.js [hunk 5]
```diff
@@ -172,6 +172,68 @@ describe( 'PasteFromOffice - filters', () => {
...
+					const html =
+					'<html>' +
+						'<head>' +
+							'<o:SmartTagType />' +
+							'<o:SmartTagType >' +
+						'</head>' +
+						'<body>' +
+							'<p>foo</p>' +
+						'</body>' +
+					'</html>';
```
Comment (niegowski) [inline packages/ckeditor5-paste-from-office/tests/filters/parse.js:208]:
```suggestion
						'<html>' +
							'<head>' +
								'<o:SmartTagType />' +
								'<o:SmartTagType >' +
							'</head>' +
							'<body>' +
								'<p>foo</p>' +
							'</body>' +
						'</html>';
```

**Comment by arkflpc** [comment]
LGTM

**Review by niegowski — APPROVED** [review]

**Pull request merged**